Realm/Realm/Handles/SessionHandle.cs

```diff
@@ -281,6 +286,7 @@ private static void HandleSessionError(IntPtr sessionHandlePtr, ErrorCode errorC
                     var clientResetEx = new ClientResetException(session.User.App, messageString, errorCode, userInfo);
 
                     if (syncConfig.ClientResetHandler is DiscardLocalResetHandler ||
+                        syncConfig.ClientResetHandler is AutomaticRecoveryHandler ||
                         syncConfig.ClientResetHandler?.ManualClientReset != null)
```

Comment:
[nit] I know that's code you didn't change, but we can remove the null propagation here
```suggestion
                        syncConfig.ClientResetHandler.ManualClientReset != null)
```